wolfcrypt/src/evp.c

```diff
@@ -8118,8 +8118,12 @@ void wolfSSL_EVP_init(void)
 
         WOLFSSL_ENTER("wolfSSL_EVP_Cipher");
 
-        if (ctx == NULL || ((src == NULL || dst == NULL) &&
-            (TRUE
+        if (ctx == NULL) {
+            WOLFSSL_MSG("Bad argument.");
+            return WOLFSSL_FATAL_ERROR;
+        }
+
+        if (TRUE
         #ifdef HAVE_AESGCM
```

Comment:
Make the check as to whether the ctx is an AEAD algorithm a new function.
Will clean up this code and potentially be useable elsewhere.

Reply:
Refactored.

Reply:
Thanks!